src/infrastructure/torrent/range_server_test.go

```
@@ -145,6 +147,60 @@ func TestServeTorrentRangeHTTPBehavior(t *testing.T) {
+			"token": {
+				file:     blockingRangeFile{reader: reader},
+				ctx:      sourceCtx,
+				cancel:   sourceCancel,
+				requests: &sync.WaitGroup{},
+			},
+		},
+	}
+
+	requestCtx, cancelRequest := context.WithCancel(context.Background())
+	defer cancelRequest()
+	request := httptest.NewRequest(http.MethodGet, "/source/token/file.mkv", nil).WithContext(requestCtx)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)